### PR TITLE
fix: add qt4-bigsur patch

### PR DESCRIPTION
**Pull request by qkdreyer**
- https://bugs.webkit.org/show_bug.cgi?id=50991
- https://github.com/phracker/MacOSX-SDKs/blob/041600eda65c6a668f66cb7d56b7d1da3e8bcc93/MacOSX11.3.sdk/usr/include/c%2B%2B/v1/cstddef#L37

### CODE REVIEW

**Comment by qkdreyer** [comment]
```
cartr/qt4/qt@4: stable 4.8.7 (bottled), HEAD
Cross-platform application and UI framework
https://www.qt.io/
/usr/local/Cellar/qt@4/4.8.7_6 (2,971 files, 76.4MB)
  Poured from bottle on 2021-08-03 at 10:17:07
/usr/local/Cellar/qt@4/4.8.7_7 (3,730 files, 85.3MB) *
  Built from source on 2021-09-03 at 16:09:02
From: https://github.com/qkdreyer/homebrew-qt4/blob/HEAD/qt@4.rb
```

File: qt@4.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ class QtAT4 < Formula
   url "https://download.qt.io/archive/qt/4.8/4.8.7/qt-everywhere-opensource-src-4.8.7.tar.gz"
   mirror "https://mirrors.ocf.berkeley.edu/qt/archive/qt/4.8/4.8.7/qt-everywhere-opensource-src-4.8.7.tar.gz"
   sha256 "e2882295097e47fe089f8ac741a95fef47e0a73a3f3cdf21b56990638f626ea0"
-  revision 6
+  revision 7
```
Comment (cartr) [inline qt@4.rb]: I'm not sure we need to bump the revision here. (Mostly I'm kinda lazy and don't want to rebuild all the bottles.)

File: qt@4.rb [hunk 2]
```diff
@@ -66,6 +66,12 @@ class QtAT4 < Formula
     sha256 "5e81df9a1c35a5aec21241a82707ad6ac198b2e44928389722b64da341260c5d"
   end
 
+  # Patch to fix build on macOS Big Sur
+  patch do
+    url "https://raw.githubusercontent.com/qkdreyer/homebrew-qt4/98a906e1ad47106c65021150938de61138799ea8/patches/qt4-bigsur.patch"
```
Comment (cartr) [inline qt@4.rb]: Could you please update this to say `/cartr/` instead of `/qkdreyer/` so it points at the original repository and not your fork?

**Review by cartr — CHANGES_REQUESTED** [review]
Thanks for submitting this PR, and apologies for the delay! I don't have a Mac handy to test this but it looks pretty reasonable.

Would you be interested in becoming a maintainer on this repository?

**Comment by qkdreyer** [comment]
Hello :) Sorry for my delayed response, I was in holidays.
I've just pushed changes you've required.
About becoming a maintainer, I guess I'm willing to try 🙃

**Comment by cartr** [comment]
Looks great, thank you for submitting this!

**Pull request merged**